### DISCUSSION TITLE
add options to hide History and Logbook

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [Mariusthvdb/custom-attributes#49] [Ideas]
as discussed on Kiosk-mode and elsewhere, it would be good to add some customization options to hide the Logbook and History sections on more-info panels, in such a way that it is not just an overall generic setting.

If possible, we should follow the recorder/history/logbook settings of a user, and only show, where an entity has history/logbook actually recorded. Like that. globally. without ay further config from a user.

<img width="574" alt="Scherm­afbeelding 2023-12-19 om 19 38 06" src="https://github.com/Mariusthvdb/custom-attributes/assets/33354141/192a46e5-6f04-4b8c-b2d8-74709aeda6ea">

its a pain to the eye to keep seeing these sections while no data is available, And HA should know about it. Well, it does know about it, because it says so in the Statistic section of developer-tools...

If that would proof to be impossible, then we should try and seek a way to delete the sections just like we do now with filtering attributes:
by_entity_id
by_domain
by_glob
by_device_class

to stay as close to current custom_attributes as possible.
These options would allow us to hide the sections, even if there would be data recorded

We probably need to rename the plugin to custom-more-info

**Comment by elchininet** [comment]
>If possible, we should follow the recorder/history/logbook settings of a user, and only show, where an entity has history/logbook actually recorded. Like that. globally. without ay further config from a user.

Remember that front-end cannot read anything from the back-end configuration, if we need to set up some configuration it should be done in front-end files (in the same place that the current configuration is placed).

**Reply by Mariusthvdb** [reply]
yes, and that would be easy?
myself, I import my recorder filters in `included_entities.yaml`

```
#recorder
# db_url: !secret db_url
purge_keep_days: 7

# https://www.home-assistant.io/integrations/recorder/#configure-filter
exclude:
  !include excluded_entities.yaml

include:
  !include included_entities.yaml
```
I suppose any frontend card could import that same file?
dont we do something like that already with the entity_attributes in https://github.com/Mariusthvdb/custom-attributes/blob/main/src/constants/index.ts ?

I suppose the relevant entries in recorder are:
```
entity_globs:

domains:

entities:
```

which coincide with what we use now in custom_attributes ;-)

But, if too complex, we could resort to the second option in my OP, and make it even possible to hide when actually available data

**Reply by elchininet** [reply]
>yes, and that would be easy?

I think that it should not be difficult to do, all the selection work is done behind the scenes by the `home-assistant-query-selector` package, what it is needed to do is just to add styles.

>I suppose any frontend card could import that same file?

No, front-end don‘t have access to import any back-end configuration, imagine the security risk that would be if any `JavaScript` code could read what you are setting up inside a Docker container 😉

What you can do is to include that file inside the lovelace dashboard `yaml` file, front-end have access to dashboards config because it is something set up in the front-end.

>dont we do something like that already with the entity_attributes in https://github.com/Mariusthvdb/custom-attributes/blob/main/src/constants/index.ts ?

No, those are just constants harcoded in the code.

You can do this, this should work I suppose:

```yaml
hide_history:
  by_entity_id: !include excluded_entities.yaml

hide_logbook:
  by_entity_id: !include excluded_entities.yaml
```

**Reply by Mariusthvdb** [reply]
But that is exactly what I meant ;-)
